packages/patternfly-3/patternfly-react/src/components/LoginPage/components/LoginCardComponents/LoginCardWithValidation.js

```diff
@@ -44,6 +44,11 @@ class LoginCardWithValidation extends React.Component {
         showError: false
       }
     });
+    this.setState(prevState => ({
```

Comment:
could you move this into the upper `setState` so it won't update the state twice?

Reply:
I am fine with that,
@jeff-phillips-18 @priley86 what is your opinion on that?

Reply:
Would prefer not to setState twice, but I understand. Best I can come up with FWIW:

```
     const updateState = {
        usernameField: {value: inputType === 'usernameField' ? e.target.value : this.state.usernameField.value},
        passwordField: {value: inputType === 'passwordField' ? e.target.value : this.state.passwordField.value}
      };
      updateState[inputType].showError = false;
      updateState.form = {
        disableSubmit: updateState.usernameField.value.length < 1 || updateState.passwordField.value.length < 1
      };
      this.setState(updateState);
 ```

Reply:
I've used this suggestion, thanks